src/lava/utils/float2fixed.py

```diff
@@ -1,26 +1,719 @@
+        self.conv_data = self._get_conv_data()
+        # Instantiate monitors for recording from dynamic variables.
+        self.monitors = self._create_monitoring_infrastructure()
+        # Run floating-point model and store activations monitored variables.
+        self._run_procs()
+        # Update scale domains for scaling of parameters.
+        self.scale_domains = self._update_scale_domains()
+        # Get the factor/ for the scaling functions for parameter conversion.
+        self.scaling_factors = self._find_scaling_factors()
+        # Scale parameter from a floating- to a fixed-point representation.
+        self.scaled_params = self.scale_parameters()
+
+    def _set_procs(self, proc, find_connected_procs=True) -> None:
+        """Set list of Lava Processes for conversion. Either pass a Lava
+        Process and find connected Lava Processes or pass list of Lava
+        Processes.
+        Write Processes p in a dictionary with structure {p.id : p}.
+
+        Parameters
+        ----------
+
+        proc : AbstracProcess or List[AbstractProcess]
+            Processes to be converted
+        find_connected_procs : bool
+            Find all Processes connected to passed Process
+        """
+        if issubclass(type(proc), AbstractProcess):
+            if find_connected_procs:
+                proc_list = find_processes(proc)
+            else:
+                proc_list = [proc]
+        elif isinstance(proc, list):
+            for p in proc:
+                if not issubclass(type(p), AbstractProcess):
+                    raise TypeError("list 'proc' must contains Processes")
+                proc_list = proc
+        else:
+            raise TypeError("'proc' must be Process of list of Processes")
+
+        # Get ProcessModel classes of Processes with fixed-point RunConfig.
+        p_model_cls = ProcGroupDiGraphs._map_proc_to_model(
+            proc_list,
+            self.floating_pt_rcfg)
+
+        # Instantiate fixed-point ProcessModel of Processes and put them into
+        # dictionary. Moreover we update the self.procs from a new proc list
+        # containing Subprocesses but not their Hierarchical Processes anymore.
+        proc_list_w_sub = []
+        for p, model_cls in p_model_cls.items():
+            proc_list_w_sub.append(p)
+
+        # Find hierarchical Processes
+        hierarchical_procs_set = set(proc_list) - set(proc_list_w_sub)
+
+        # Updated Processes used for covnersion and set member storing
+        # hierarchical Processes.
+        self.procs = dict([(p.id, p) for p in proc_list_w_sub])
+        self.hierarchical_procs = dict([(p.id, p) for p in
+                                        hierarchical_procs_set])
+
+        self.run_proc = proc_list[0]
+
+    def _explicate_hierarchical_procs(self):
+        """Given a Model with hierarchical Processes, set up an equivalent
+        model with only first-order Processes, i.e. not hierarchical Processes.
```

Comment:
PReviously, we called this leaf processes

Reply:
Renamed in 58fd18a6c5ec776b81b991a6260f4edcc2d2ca2d.